ddl/ddl.go

```diff
@@ -193,9 +193,10 @@ type ddl struct {
 	limitJobCh chan *limitJobTask
 
 	*ddlCtx
-	workers     map[workerType]*worker
-	sessPool    *sessionPool
-	delRangeMgr delRangeManager
+	workers           map[workerType]*worker
+	sessPool          *sessionPool
+	delRangeMgr       delRangeManager
+	enableTiFlashPoll bool
```

Comment:
This `enableTiFlashPoll` is wrapped by `IsTiFlashPollEnabled`, and used in `AlterTableSetTiFlashReplica`/`AddTablePartitions`.

Meanwhile in the current implementation, enableTiFlashPoll is by default true,  which can break some old tests. `IsTiFlashPollEnabled` helps that.

Reply:
> The typical way is allocate a dedicated session for the background service. It is not good idea to get the session again and again.

Can I have an example on how to do so?

Reply:
> 



> This `enableTiFlashPoll` is wrapped by `IsTiFlashPollEnabled`, and used in `AlterTableSetTiFlashReplica`/`AddTablePartitions`.
> 
> Meanwhile in the current implementation, enableTiFlashPoll is by default true, which can break some old tests. `IsTiFlashPollEnabled` helps that.

So my suggestion is like remove that. Wheter tiflash is enabled or not, you construct the rules. And `AlterTableSetTiFlashReplica` will either use a mocking manager or a real manager, which should be determined by `enableTiFlashPoll`. It is somehow similar to the `delRangeManager`, check 
https://github.com/pingcap/tidb/blob/dc9e75e0098648fdedab3ebc97a8f8b01190e8fe/ddl/ddl.go#L334-L340

Reply:
> > The typical way is allocate a dedicated session for the background service. It is not good idea to get the session again and again.
> 
> Can I have an example on how to do so?

If you mean 5m polling, it does not matter much to get the session again and again.

There is no direct example to create session only once in ddl, there are examples here, though, check this:

https://github.com/pingcap/tidb/blob/master/session/session.go#L2625-L2639

If you meant to add something polling forever, I actually recommended you add it in domain instead of ddl. In DDL, you could get the session and never return it until the end.

Reply:
> It is somehow similar to the delRangeManager

Do you mean wrap logics into a tiflashManager?